lib/ansible/modules/windows/win_regedit.ps1

```
@@ -21,313 +22,326 @@
+    if ($string.Length -eq 1) {
+        return ,@([System.Convert]::ToByte($string,16))
+    } elseif (($string.Length % 2 -eq 0) -and ($string.IndexOf(":") -eq -1)) {
+        return ,@($string -split '([a-f0-9]{2})' | foreach-object { if ($_) {[System.Convert]::ToByte($_,16)}})
+    } elseif ($string.IndexOf(":") -ne -1) {
+        return ,@($string -split ':+' | foreach-object {[System.Convert]::ToByte($_,16)})
     } else {
         return ,@()
     }
 }
 
-# Create the required PSDrives if missing
-if (-not (Test-Path HKCR:\)) {
-    New-PSDrive -Name HKCR -PSProvider Registry -Root HKEY_CLASSES_ROOT
-}
-if (-not (Test-Path HKU:\)) {
-    New-PSDrive -Name HKU -PSProvider Registry -Root HKEY_USERS
-}
-if (-not (Test-Path HKCC:\)) {
-    New-PSDrive -Name HKCC -PSProvider Registry -Root HKEY_CURRENT_CONFIG
+Function Get-ValueType($type) {
+    switch ($type) {
+        "none" { [Microsoft.Win32.RegistryValueKind]::None }
+        "binary" { [Microsoft.Win32.RegistryValueKind]::Binary }
+        "dword" { [Microsoft.Win32.RegistryValueKind]::DWord }
+        "expandstring" { [Microsoft.Win32.RegistryValueKind]::ExpandString }
+        "multistring" { [Microsoft.Win32.RegistryValueKind]::MultiString }
+        "string" { [Microsoft.Win32.RegistryValueKind]::String }
+        "qword" { [Microsoft.Win32.RegistryValueKind]::QWord }
+    }
 }
 
-
-# Convert HEX string to binary if type binary
-if ($type -eq "binary" -and $data -ne $null -and $data -is [string]) {
-    $data = Convert-RegExportHexStringToByteArray($data)
+Function Test-RegistryProperty($path, $name) {
+    # will validate if the registry key contains the property, returns true
+    # if the property exists and false if the property does not
+    try {
+        $value = (Get-Item -Path $path).GetValue($name)
+        if ($value -eq $null) {
+            return $false
+        } else {
+            return $true
+        }
+    } catch {
```

Should probably explicitly catch ItemNotFoundException, not blanket swallow all exceptions